src/schedule/zephyr_ll.c

```diff
@@ -84,6 +84,39 @@ static void zephyr_ll_task_insert_unlocked(struct zephyr_ll *sch, struct task *t
 		list_item_append(&task->list, &sch->tasks);
 }
 
+/* perf measurement windows size 2^x */
+#define CYCLES_WINDOW_SIZE	10
+
+static inline enum task_state do_task_run(struct task *task)
+{
+	uint32_t cycles0, cycles1, diff;
+	enum task_state state;
+
+	cycles0 = k_cycle_get_32();
+
+	state = task_run(task);
+
+	cycles1 = k_cycle_get_32();
+	if (cycles1 > cycles0)
+		diff = cycles1 - cycles0;
+	else
+		diff = UINT32_MAX - cycles0 + cycles1;
+
```

Comment:
Tempted to just ignore rollover of the cycle counter, but ended up adding this to the patch.